src/vs/workbench/contrib/terminal/browser/terminalQuickAccess.ts

```diff
@@ -32,8 +37,26 @@ export class TerminalQuickAccessProvider extends PickerQuickAccessProvider<IPick
 			const terminalGroup = terminalGroups[groupIndex];
 			for (let terminalIndex = 0; terminalIndex < terminalGroup.terminalInstances.length; terminalIndex++) {
 				const terminal = terminalGroup.terminalInstances[terminalIndex];
-				const label = `$(${terminal.icon?.id}) ${groupIndex + 1}.${terminalIndex + 1}: ${terminal.title}`;
-
+				const icon = terminal.icon;
+				const iconId = this._getIconId(icon);
+				const label = `$(${iconId}) ${groupIndex + 1}.${terminalIndex + 1}: ${terminal.title}`;
+				const iconClasses: string[] = [];
+				const color = terminal.color ? `terminal-icon-${terminal.color}` : ThemeIcon.isThemeIcon(icon) ? `terminal-icon-${icon.color?.id}`.replace('.', '_') : undefined;
+				if (color) {
+					iconClasses.push(color);
+				}
+				let uri = undefined;
+				if (icon instanceof URI) {
+					uri = icon;
+				} else if (icon instanceof Object && 'light' in icon && 'dark' in icon) {
+					uri = this._themeService.getColorTheme().type === ColorScheme.LIGHT ? icon.light : icon.dark;
+				}
+				if (uri instanceof URI) {
+					const uriIconKey = hash(uri.path).toString(36);
+					const className = `terminal-uri-icon-${uriIconKey}`;
+					iconClasses.push(className);
+					iconClasses.push(`terminal-uri-icon`);
```

Comment:
We should consolidate getting the icon classes somewhere. Maybe a terminalIcon.ts file with a `TerminalIcon` namespace with helpers like extracting the classes/hash/etc.?